src/SFA.DAS.EmployerIncentives.Events/EarningsResilienceCheck/EarningsCalculationRequiredHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Abstractions.Events;
+using SFA.DAS.EmployerIncentives.Commands.Types.ApprenticeshipIncentive;
+using SFA.DAS.EmployerIncentives.Domain.EarningsResilienceCheck.Events;
+using SFA.DAS.EmployerIncentives.Domain.IncentiveApplications.Events;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Events.EarningsResilienceCheck
+{
+    public class EarningsCalculationRequiredHandler : IDomainEventHandler<EarningsCalculationRequired>
+    {
+        private readonly ICommandPublisher _commandPublisher;
+
+        public EarningsCalculationRequiredHandler(ICommandPublisher commandPublisher)
+        {
+            _commandPublisher = commandPublisher;
+        }
+
+        public Task Handle(EarningsCalculationRequired @event, CancellationToken cancellationToken = default)
+        {
+            var commands = new List<Task>();
+            foreach (var apprenticeship in @event.Model.ApprenticeshipModels)
+            {
+                var command = new CreateApprenticeshipIncentiveCommand(
+                    @event.Model.AccountId,
+                    @event.Model.AccountLegalEntityId,
+                    apprenticeship.Id,
+                    apprenticeship.ApprenticeshipId,
+                    apprenticeship.FirstName,
+                    apprenticeship.LastName,
+                    apprenticeship.DateOfBirth,
+                    apprenticeship.Uln,
+                    apprenticeship.PlannedStartDate,
+                    apprenticeship.ApprenticeshipEmployerTypeOnApproval
+                );
+
+                var task = _commandPublisher.Publish(command);
```

Comment:
What happens here is the apprenticeship incentive exists but it doesn't have earnings?  A quick look suggests it'll create a duplicate apprenticeship incentive record.

Reply:
We need to do one of two things here...
1) Change CreateApprenticeshipIncentiveCommand so it doesn't create a duplicate record but does publish the Created command.
2) Change this implementation to check for the existence of the ApprenticeshipIncentive and if it doesn't exist publish the CreateApprenticeshipIncentiveCommand but if it does exist then publish a CalculateEarningsCommand

Reply:
This still looks like a big issue to me?